rerun_notebook/src/js/widget.ts

```diff
@@ -0,0 +1,107 @@
+import { LogChannel, Panel, PanelState, WebViewer } from "@rerun-io/web-viewer";
+import type { AnyModel, Render } from "@anywidget/types";
+import "./widget.css";
+
+type PanelStates = Partial<Record<Panel, PanelState>>;
+
+const PANELS = ["top", "blueprint", "selection", "time"] as const;
+
+/* Specifies attributes defined with traitlets in ../rerun_notebook/__init__.py */
+interface WidgetModel {
+  width?: number;
+  height?: number;
+
+  _url?: string;
+  _panel_states?: PanelStates;
+}
+
+type Opt<T> = T | null | undefined;
+
+class ViewerWidget {
```

Comment:
The widget here is a wrapper over our existing `WebViewer` package. It sets up a few callbacks to listen for changes to `url`, `panel_states`, and incoming data which is forwarded to the viewer using a temporary `LogChannel`.